matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/DefaultSession.kt

```diff
@@ -297,4 +307,10 @@ internal class DefaultSession @Inject constructor(
     override fun logDbUsageInfo() {
         RealmDebugTools(realmConfiguration).logInfo("Session")
     }
+
+    override fun launch(block: suspend () -> Unit) {
```

Comment:
Naming is a bit light... Maybe also you should at least add context param like the regular launch method.

Reply:
> Naming is a bit light... 

I like this naming, can replace `viewModelScope.launch { ... }` by `session.launch { ... }`, this is nice
 
> Maybe also you should at least add context param like the regular launch method.

Done

Reply:
This should probably return a `Job`.
A potential issue here is, are you going to whack-a-mole every single coroutine creating function? `launch`, `async`, `produce`, `launchIn`, etc.

Maybe this should be a `withContext` kinda thing that throws `CancellationException` when the session is not active?